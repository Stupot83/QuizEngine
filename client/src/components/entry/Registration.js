import React, { Component } from "react";
import { Link } from "react-router-dom";
import "../../sass/Registration.scss";
import Button from "@material-ui/core/Button";
import CreateIcon from "@material-ui/icons/Create";

class Registration extends Component {
    constructor() {
        super();
        this.state = {
            name: "",
            email: "",
            password: "",
            password2: "",
            invalidEntries: {}
        };
    }

    onChange = e => {
        this.setState({ [e.target.id]: e.target.value });
    };

    onSubmit = e => {
        e.preventDefault();

        const newUser = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password,
            password2: this.state.password2
        };
    };

    render() {
        const { invalidEntries } = this.state;

        return (
            <div className="Registration_container">
                <div id="Registration_header" className="Registration_header">
                    Register for QuizEngine
                </div>

                <form className="Registration_form" noValidate onSubmit={this.onSubmit}>
                    <div className="Registration_details">
                        <label>
                            <div id="Name_field" className="Registration_label">
                                Name
                            </div>
                            <input
                                onChange={this.onChange}
                                value={this.state.name}
                                error={invalidEntries.name}
                                id="name"
                                type="text"
                                autoComplete="off"
                                className="Registration_field"
                            />
                            <div className="Registration_invalid_entry">{invalidEntries.name}</div>
                        </label>
                    </div>

                    <div className="Registration_details">
                        <label>
                            <div id="Email_field" className="Registration_label">
                                Email
                            </div>
                            <input
                                onChange={this.onChange}
                                value={this.state.email}
                                error={invalidEntries.email}
                                id="email"
                                type="email"
                                autoComplete="off"
                                className="Registration_field"
                            />
                            <div className="Registration_invalid_entry">{invalidEntries.email}</div>
                        </label>
                    </div>

                    <div className="Registration_details">
                        <label>
                            <div id="Password_field" className="Registration_label">
                                Password
                            </div>
                            <input
                                onChange={this.onChange}
                                value={this.state.password}
                                error={invalidEntries.password}
                                id="password"
                                type="password"
                                autoComplete="off"
                                className="Registration_field"
                            />
                            <div className="Registration_invalid_entry">{invalidEntries.password}</div>
                        </label>
                    </div>

                    <div className="Registration_details">
                        <label>
                            <div id="Confirm_password_field" className="Registration_label">
                                Confirm Password
                            </div>
                            <input
                                onChange={this.onChange}
                                value={this.state.password2}
                                error={invalidEntries.password2}
                                id="password2"
                                type="password"
                                autoComplete="off"
                                className="Registration_field"
                            />
                            <div className="Registration_invalid_entry">{invalidEntries.password2}</div>
                        </label>
                    </div>

                    <div className="Registration_button_box">
                        <Button
                            id="Registration_button"
                            type="submit"
                            variant="contained"
                            size="large"
                            className="Registration_button"
                            startIcon={<CreateIcon />}
                        >
                            Register
                        </Button>
                    </div>
                    <div className="Registration_bottom">
                        <p id="Redirect_message">
                            Already registered for QuizEngine?{" "}
                            <Link to="/login" id="Login_link" className="Registration_login_link">
                                Login
                            </Link>
                        </p>
                    </div>
                </form>
            </div>
        );
    }
}

export default Registration;
