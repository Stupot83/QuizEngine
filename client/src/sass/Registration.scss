.Registration_container {
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
 height: 100vh;
 background: #e6e6e6;
 margin-top: -64px;
 background-color: ghostwhite;
 background-image: url("https://www.transparenttextures.com/patterns/brick-wall.png");

 .Registration_header {
  color: #D1345B;
  padding: 2rem;
  font-size: 20px;
  text-align: center;
  font-weight: bold;
  margin-bottom: 20px;
 }
 
 .Registration_form {
  width: 500px;
  background-color: ghostwhite;
  border: 1px solid ghostwhite;
  padding: 2rem;
 }

 .Registration_details {
  margin-bottom: 1rem;

  .Registration_label {
   width: 100%;
   padding-right: 2rem;
   margin-bottom: 0.75rem;
   font-size: 14px;
  }
  
  .Registration_field {
   background-color: white;
   border-bottom: 1px solid #3454d1;
   height: 40px;
   width: 90%;
   line-height: 1.5em;
   outline: 0;
   transition: all 0.25s ease 0s;
   font-size: 16px;
   padding: 0 1rem;
  
   &:hover {
       transition: all 0.25s ease 0s;
       background-color: #EFEFEF;
   }
  
   &:focus {
       transition: all 0.25s ease 0s;
       background-color: #EFEFEF;
   }
  }
  
  .Registration_invalid_entry {
   margin-top: 0.5rem;
   color: red;
   font-size: 14px;
  }
 }

 .Registration_button_box {
  text-align: center;
  margin: 40px;

  .Registration_button {
   min-width: 150px;
    margin-left: 30px;
    margin-right: 30px;
    padding: 20px;
    border-radius: 15px;
    background-color: #D1345B;
    color: ghostwhite;
    font-weight: bold;
  }
 }
 
 .Registration_bottom {
  display: flex;
  justify-content: space-around;
  margin-top: 2rem;

  .Registration_login_link {
   text-decoration: none !important;
   color: #3454d1;
  }
 }

 @media (max-width: 800px) {
  .Registration_bottom {
      flex-direction: column;
      padding-left: 0px;
      text-align: center;
  }
 
  .Registration_form {
      width: 80%;
  }
 } 
}
