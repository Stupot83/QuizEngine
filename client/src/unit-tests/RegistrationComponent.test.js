import React from "react";
import { shallow } from "enzyme";
import Registration from "../components/entry/Registration";
import QuizEngineStore from "../quizEngineStore";
import { Provider } from "react-redux";

const mockStore = QuizEngineStore;

describe("Registration component", () => {
    it("should render correctly in debug mode", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Registration debug />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Registration />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
