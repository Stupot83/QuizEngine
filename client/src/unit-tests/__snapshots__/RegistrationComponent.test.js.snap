// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component should render correctly 1`] = `ShallowWrapper {}`;

exports[`Registration component should render correctly in debug mode 1`] = `ShallowWrapper {}`;
